client/src/api/user/index.ts

```diff
@@ -46,6 +47,17 @@ const updateUser = async (
   return await put(url, profileData, failureMessage);
 };
 
+// Update user's onboarding status
+const updateOnboardingStatus = async (
+  userId: string,
+  status: keyof typeof onboardingStatusEnum,
+): Promise<Response<UserResponse>> => {
+  const url = buildEndpoint(USER_ENDPOINT, userId);
+  const failureMessage = 'UPDATE_USER_FAIL';
+  const profileData = { onboardingStatus: status };
+  return await put(url, profileData, failureMessage);
```

Comment:
NIT: could you add a line break just to be consistent with other functions and so it's a little more readable (less dense)